prefect_snowflake/database.py

```diff
@@ -53,32 +54,8 @@ def _get_connect_params(self) -> Dict[str, str]:
             "database": self.database,
             "warehouse": self.warehouse,
             "schema": self.schema_,
-            # required to track task's usage in the Snowflake Partner Network Portal
-            "application": "Prefect_Snowflake_Collection",
-            **self.credentials.dict(),
-        }
-
-        # filter out unset values
-        connect_params = {
-            param: value for param, value in connect_params.items() if value is not None
         }
 
-        for param in ("password", "private_key", "token"):
-            if param in connect_params:
-                connect_params[param] = connect_params[param].get_secret_value()
-
-        # set authenticator to the actual okta_endpoint
-        if connect_params.get("authenticator") == "okta_endpoint":
-            endpoint = connect_params.pop("endpoint", None) or connect_params.pop(
-                "okta_endpoint", None
-            )  # okta_endpoint is deprecated
-            connect_params["authenticator"] = endpoint
-
-        private_der_key = self.credentials.resolve_private_key()
-        if private_der_key is not None:
-            connect_params["private_key"] = private_der_key
-            connect_params.pop("password", None)
-
         return connect_params
 
     def get_connection(self) -> snowflake.connector.SnowflakeConnection:
```

Comment:
Maybe this can accept `connect_kwargs` too? And also add a docstring if so
```suggestion
    def get_connection(self, **connect_kwargs: Dict[str, Any]) -> snowflake.connector.SnowflakeConnection:
```